synapse/events/utils.py

```diff
@@ -490,9 +494,10 @@ def _inject_bundled_aggregations(
             serialized_aggregations[RelationTypes.REFERENCE] = aggregations.references
 
         if aggregations.replace:
-            # If there is an edit, apply it to the event.
+            # If there is an edit, optionally apply it to the event.
             edit = aggregations.replace
-            self._apply_edit(event, serialized_event, edit)
+            if apply_edits:
```

Comment:
This is fine, but I think I would have combined it into the if-statement above.

Reply:
... but then the `serialized_aggregations` update would be missed?

Reply:
Ah, I misread the indentation, yeah this looks good! 👍